Description: Extend effect of USE_PACKAGED option
 Enable the DESKTOP_APP_USE_PACKAGED option for the tl-expected,
 the qr_code_gen, and the mapbox-variant libraries.
Author: Nicholas Guriev <guriev-ns@ya.ru>
Date: Mon, 27 Jan 2020 08:58:41 +0300

--- a/cmake/external/expected/CMakeLists.txt
+++ b/cmake/external/expected/CMakeLists.txt
@@ -7,6 +7,12 @@
 add_library(external_expected INTERFACE IMPORTED GLOBAL)
 add_library(desktop-app::external_expected ALIAS external_expected)
 
+if (DESKTOP_APP_USE_PACKAGED)
+    find_package(tl-expected REQUIRED)
+    target_link_libraries(external_expected INTERFACE tl::expected)
+    return()
+endif()
+
 target_include_directories(external_expected SYSTEM
 INTERFACE
     ${third_party_loc}/expected/include
--- a/cmake/external/qr_code_generator/CMakeLists.txt
+++ b/cmake/external/qr_code_generator/CMakeLists.txt
@@ -4,8 +4,14 @@
 # For license and copyright information please follow this link:
 # https://github.com/desktop-app/legal/blob/master/LEGAL
 
-add_library(external_qr_code_generator OBJECT)
+add_library(external_qr_code_generator INTERFACE IMPORTED GLOBAL)
 add_library(desktop-app::external_qr_code_generator ALIAS external_qr_code_generator)
+
+if (DESKTOP_APP_USE_PACKAGED)
+    target_include_directories(external_qr_code_generator INTERFACE /usr/include/qrcodegen)
+    target_link_libraries(external_qr_code_generator INTERFACE qrcodegencpp)
+    return()
+endif()
 init_target(external_qr_code_generator "(external)")
 
 set(qr_loc ${third_party_loc}/QR)
--- a/cmake/external/variant/CMakeLists.txt
+++ b/cmake/external/variant/CMakeLists.txt
@@ -6,6 +6,9 @@
 
 add_library(external_variant INTERFACE IMPORTED GLOBAL)
 add_library(desktop-app::external_variant ALIAS external_variant)
+if (DESKTOP_APP_USE_PACKAGED)
+    return()
+endif()
 
 target_include_directories(external_variant SYSTEM
 INTERFACE
